pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1466,7 +1480,10 @@ public void seek(long timestamp) throws PulsarClientException {
             seekFuture.complete(null);
         }).exceptionally(e -> {
             log.error("[{}][{}] Failed to reset subscription: {}", topic, subscription, e.getCause().getMessage());
-            seekFuture.completeExceptionally(e.getCause());
+            seekFuture.completeExceptionally(
+                new PulsarClientException.WrapperException(
```

Comment:
I am not sure WrapperException is the right fix here. because it will change the client behavior. for example, if a client used to handle exeception A, now we are throwing a WrapperException of A.

We should reconstruct a A with an updated error message and its stack trace.

Reply:
because there are some methods when it uses the exception it will use the `unwrap` method to get the real exception.

Reply:
But it seems if I want to add the error message there are two ways to update. One is updating the error message before the exception throws. Another is unwrapping the wrapper when the method needs to handle the exception.